### DISCUSSION TITLE
Provide OpenApi definition of https bindings

### DISCUSSION THREAD

**Discussion by juliapampus** [International-Data-Spaces-Association/ids-specification#155] [Ideas]
`catalog.binding.https.md` says:
> The OpenAPI definitions for this specification can be accessed [here](TBD).

Would be a nice idea to have an OpenApi definition of the HTTPS bindings.

Same goes for negotiation and transfer.

**Comment by sebbader-sap** [comment]
Alternative for OpenApi for the asynchronous operations: https://www.asyncapi.com (@juliapampus you were right with the link)

**Comment by matgnt** [comment]
I've created a first draft version of an openapi spec for the catalog part:
https://github.com/boschresearch/py-cx-ids/blob/dev/pycxids/core/http_binding/http_binding_openapi.yaml

Since I didn't find tooling to generate the models from existing context / ttl files, I did this manually.
I maninly used this catalog example:
https://github.com/International-Data-Spaces-Association/ids-specification/blob/main/catalog/message/catalog.json

I guess this is very error-prone and should be automated to be in sync.

---
Matthias Binzer
